pkg/generator/certificate.go

```diff
@@ -425,6 +424,8 @@ func NewRootCA(keyConfig *v1alpha1.KeyConfig) (*CertKeyPair, error) {
 		Cert: &Certificate{},
 	}
 	rCA.V1Spec = keyConfig.Spec
-	rCA.V1Spec.Algorithm = v1alpha1.AlgorithmTypeECDSAWithSHA256
+	if rCA.V1Spec.Algorithm == "" {
+		rCA.V1Spec.Algorithm = v1alpha1.AlgorithmTypeECDSAWithSHA256
+	}
```

Comment:
We already handle this in the webhook change above. The algorithm will already be set by the time we get here.
```suggestion
```

Reply:
Yep - I did that on purpose. we might get a null pointer on this if the webhooks get skipped

Reply:
it shouldn't 🤔 unless you're running debugging/running locally.